src/docopt/dispatch.nim

```diff
@@ -0,0 +1,69 @@
+import docopt, macros, strutils, sequtils, typetraits
+
+macro runUserImplemented(x: typed, fallback: untyped): untyped =
+  let typ = x.getType
+  if typ.kind == nnkBracketExpr and typ[0].kind == nnkSym and $typ[0] == "typeDesc":
```

Comment:
Already discussed on IRC, but for completeness: This currently creates code for a call to a `to<CustomType>` proc, even if that doesn't exist. As we agreed it should really emit a `when declared(to<CustomType>)` as a guard.

edit: and mention that this is the magic macro that performs this in a short docstring.